test/sensor.js

```diff
@@ -103,132 +143,612 @@ exports["Sensor - Analog"] = {
+    // this.clock.tick(tickDelta);
+    // tickAccum += tickDelta;
+    // test.strictEqual(dataSpy.callCount, 4, "tick " + tickAccum + ": data event handler should be called every multiple of " + this.defShape.freq + " ticks");
+    // test.strictEqual(limitSpy.callCount, 2, "tick " + tickAccum + ": should not call limit event handler at tick " + this.defShape.freq * 4 + "; on lower limit");
+    // test.strictEqual(lowerSpy.callCount, 1, "tick " + tickAccum + ": should not call limit:lower event handler at tick " + this.defShape.freq * 4 + "; on lower limit");
+    // test.strictEqual(upperSpy.callCount, 1, "tick " + tickAccum + ": should not call limit:upper event handler at tick " + this.defShape.freq * 4 + "; on lower limit");
+    //
+    // tickDelta = this.defShape.freq - 1;
+    // this.clock.tick(tickDelta);
+    // tickAccum += tickDelta;
+    // raw = 550;
+    // callback(raw);
+    //
+    // test.strictEqual(dataSpy.callCount + limitSpy.callCount + lowerSpy.callCount + upperSpy.callCount, 8,
+    //   "tick " + tickAccum + ": no event handlers should be called again until tick " + this.defShape.freq * 5);
+    //
+    // tickDelta = 1;
+    // this.clock.tick(tickDelta);
+    // tickAccum += tickDelta;
+    // test.strictEqual(dataSpy.callCount, 5, "tick " + tickAccum + ": data event handler should be called every multiple of " + this.defShape.freq + " ticks");
+    // test.strictEqual(limitSpy.callCount, 2, "tick " + tickAccum + ": should not call limit event handler at tick " + this.defShape.freq * 5 + "; on upper limit");
+    // test.strictEqual(lowerSpy.callCount, 1, "tick " + tickAccum + ": should not call limit:lower event handler at tick " + this.defShape.freq * 5 + "; on upper limit");
+    // test.strictEqual(upperSpy.callCount, 1, "tick " + tickAccum + ": should not call limit:upper event handler at tick " + this.defShape.freq * 5 + "; on upper limit");
+
+    test.done();
+  },// ./limit: function(test)
+
+  freq: function(test) {
+    var spy = sinon.spy(),
+      newShape, newFreq, tickDelta, tickAccum;
+    test.expect(10);
+
+    this.sensor.on("data", spy);
+    test.deepEqual(this.defShape, getShape(this.sensor), "sensor instance properties should match default shape values");
+
+    tickAccum = 0;
+    tickDelta = this.defShape.freq - 1;
+    this.clock.tick(tickDelta);
+    tickAccum += tickDelta;
+    test.strictEqual(spy.callCount, 0, "tick " + tickAccum + ": data event handler should not be called first time until tick " + this.defShape.freq);
+
+    tickDelta = 1;
+    this.clock.tick(tickDelta);
+    tickAccum += tickDelta;
+    test.strictEqual(spy.callCount, 1, "tick " + tickAccum + ": data event handler should be called first time at tick " + this.defShape.freq);
+
+    newShape = deepCopy(this.defShape);
+    newFreq = 35;
+    newShape.freq = newFreq;
+    this.sensor.freq = newFreq;
+    test.deepEqual(newShape, getShape(this.sensor), "sensor instance properties should match shape with new freq");
```

Comment:
It verifies that the data event does not get triggered before the default this.freq ms, that it does get triggered at the default time, that setting a new freq value changes (only the freq property value of) the instance state.  Following checks verify that future data events are triggered at the new interval, offset from when the freq value was changed.